### PR TITLE
Add tests for `__str__` and `to_latex()` for Measurement and PauliRotation

### LINKED ISSUE #167: Add tests __str__ and to_latex tests for the Pauli rotation classes
Issue Description
-----------------

See #165

Proposed Solution
-----------------

Test these methods comparing outputs

### PR DESCRIPTION
**Pull request by alexnguyenn**
Closes #167. Also switch to using f-string for those functions.

### CODE REVIEW

File: tests/pauli_rotations/rotation_test.py [hunk 1]
```diff
@@ -91,6 +91,37 @@ def test_pauli_rotation_eq(self, pauli_rotation_1, pauli_rotation_2):
...
+    )
+    def test_str(self, input, expected):
+        assert str(input) == expected
+
+    @pytest.mark.parametrize(
+        "input, expected",
+        [
+            (PauliRotation.from_list([Y], Fraction(1, 2)), r"(Y)_\frac{\pi}{2}"),
+            (PauliRotation.from_list([Z, I], Fraction(-0, 4)), r"(Z \otimes I)_0"),
+            (PauliRotation.from_list([X, Y], Fraction(-6, 3)), r"(X \otimes Y)_-2\pi"),
```
Comment (gwwatkin) [inline tests/pauli_rotations/rotation_test.py]: On mobile, can't run the tests, but shouldn't shouldn't the index script be `_{-2*pi}` looks like we might have a bug

Comment (gwwatkin) [reply tests/pauli_rotations/rotation_test.py]: Since the ci passes

**Review by gwwatkin — COMMENTED** [review]
Away from my computer, but I see some strange index behavior bin the tests. It's probably due to some pre existing bug

File: tests/pauli_rotations/rotation_test.py (hunk 1 above)
Comment (alexnguyenn) [reply tests/pauli_rotations/rotation_test.py]: I just checked with overleaf. It should be `(X \otimes Y)_{-2\pi}` (which will produce something like below) so we are having a bug here.

image

Comment (alexnguyenn) [reply tests/pauli_rotations/rotation_test.py]: Just pushed a fix

Comment (gwwatkin) [reply tests/pauli_rotations/rotation_test.py]: I think its good now

File: src/lsqecc/pauli_rotations/rotation.py [hunk 2]
```diff
@@ -208,7 +205,8 @@ def __init__(self, no_of_qubit: int, isNegative: bool = False) -> None:
         self.isNegative: bool = isNegative
 
     def __str__(self) -> str:
-        return "{}M: {}".format("-" if self.isNegative else "", self.ops_list)
+        sign = "-" if self.isNegative else ""
+        return f"{sign}M: {self.ops_list}"
```
Comment (gwwatkin) [inline src/lsqecc/pauli_rotations/rotation.py:209]: This does look cleaner

**Review by gwwatkin — APPROVED** [review]
:+1:

**Pull request merged**